apps/services/views.py

```
@@ -0,0 +1,53 @@
+            return queryset.filter(vendor=user.vendor_profile)
+        return queryset.filter(is_active=True, vendor__is_active=True)
+
+    def perform_create(self, serializer):
+        """Ensure vendors can only create services for themselves."""
+
+        user = self.request.user
+        vendor = serializer.validated_data.get('vendor')
+
+        if hasattr(user, 'vendor_profile'):
+            if vendor and vendor != user.vendor_profile:
+                raise PermissionDenied('You can only manage your own services.')
```

Ruff (0.14.0)

Avoid specifying long messages outside the exception class

(TRY003)